src/rt/rwlock.rs

```diff
@@ -0,0 +1,269 @@
+use crate::rt::{
+    object::{self, Object},
+    thread, Access, Synchronize,
+};
+
+use std::collections::HashSet;
+use std::sync::atomic::Ordering::{Acquire, Release};
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub(crate) struct RwLock {
+    obj: Object,
+}
+
+#[derive(Debug, PartialEq)]
+enum RwLockType {
+    Read(HashSet<thread::Id>),
+    Write(thread::Id),
+}
+
+#[derive(Debug)]
+pub(super) struct State {
+    /// If the rwlock should establish sequential consistency.
+    seq_cst: bool,
+
+    /// `Some` when the rwlock is in the locked state.
+    /// The `thread::Id` references the thread that currently holds the rwlock.
```

Comment:
I think this comment could be made clearer, since the `lock` value could either be a single thread ID holding the write lock, or multiple thread IDs holding read locks.